### ISSUE TITLE
Compile issue AlmaLinux

### ISSUE THREAD

**Issue by pkanninen** [juanmanzanero/fastest-lap#8]
Hi,

I faced compile issue in AlmaLinux (rhel8). CMakeError.log file is in attached:

[CMakeError.log](https://github.com/juanmanzanero/fastest-lap/files/8549228/CMakeError.log)

Thanks. 
BR/Pekka

**Comment by juanmanzanero** [comment]
Hello! It looks like you are missing libpthread .

Before attempting to compile any file, the system checks your system, and this fails. What compiler are you using?

**Comment by pkanninen** [comment]
Hi,
Thanks for quick reply!
I have this library in /usr/lib64/libpthread 
Compiler is gcc  8.5.0

**Comment by juanmanzanero** [comment]
Than this means you don't have /usr/lib64 in your LD_LIBRARY_PATH, try:

`export LD_LIBRARY_PATH=/usr/lib64:${LD_LIBRARY_PATH}`

then cmake etc

**Comment by pkanninen** [comment]
Unfortunately this does not helps, still same error message

**Comment by juanmanzanero** [comment]
How did you install the compiler, via apt-get?

**Comment by pkanninen** [comment]
By: dnf group install "Development Tools"   i.e. it is same as: dnf install gcc

**Comment by juanmanzanero** [comment]
Maybe try apt-get?

**Comment by pkanninen** [comment]
I have not apt available, I try to avoid to mess two different package manager.

**Comment by juanmanzanero** [comment]
I am afraid I cannot be of much help. It looks that your development tools are not properly setup. If you have any other problem feel free to contact me

**Comment by pkanninen** [comment]
I will try find the solution for this, still thank you for help

**Comment by juanmanzanero** [comment]
Hello, can I close this issue, then?

**Comment by pkanninen** [comment]
Issue seems related to cmake linker flags. In attached CMakeError log files shows the same error what is in the following thread:
https://github.com/riebl/artery/issues/103

/usr/bin/ld: cannot find -lpthreads
[CMakeError.log](https://github.com/juanmanzanero/fastest-lap/files/8564860/CMakeError.log)
[CMakeError.log](https://github.com/juanmanzanero/fastest-lap/files/8564867/CMakeError.log)

**Comment by juanmanzanero** [comment]
It looks that you are using /usr/bin/cc. Is this gcc?

**Comment by pkanninen** [comment]
To my understanding cc linking gcc. 
Do you have ideas how I can add following cmake sets to gtest and lion compile options? 

SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pthread")
SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -pthread")

**Comment by juanmanzanero** [comment]
Honestly, I don't now how to fix your special setup. I am only familiar with compilers from apt-get, sorry.

**Comment by juanmanzanero** [comment]
Sorry, when I told you that you should put /usr/lib64 in the `LD_LIBRARY_PATH`, it should've been in the `LIBRARY_PATH`

Also, in CMake there's the option find_package(Threads)

Please give me an update on this so that we can close the issue

**Comment by pkanninen** [comment]
No succeed, exact similar end results as earlier

**Comment by juanmanzanero** [comment]
What does find_package(Threads REQUIRED) says?

**Comment by pkanninen** [comment]
> -- Configuration of fastest-lap
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
CMake Error at cmake/get-third-party.cmake:36 (find_package):
  Could not find a package configuration file provided by "lion" with any of
  the following names:

I think that first problem is "lion" does not compiled.
In attachment is full compile log file log.log
[log.log](https://github.com/juanmanzanero/fastest-lap/files/8637539/log.log)
and then gtest's CMakeOutput.log and CmakeError.log

[CMakeOutput.log](https://github.com/juanmanzanero/fastest-lap/files/8637540/CMakeOutput.log)
[CMakeError.log](https://github.com/juanmanzanero/fastest-lap/files/8637541/CMakeError.log)

**Comment by juanmanzanero** [comment]
So the program installs lots of things. Here's what it's installing and its status:

- GTest [OK]
- Lion [OK]
    - Tinyxml [OK]
    - Lapack [OK]
    - Mumps [OK]
    - Ipopt [OK]
    - Loggercpp [OK]
    - CppAD [OK]
   
So, Lion it's perfectly compiled and setup. I think now the problem it's that for some reason it is not finding it.

Can you update line 36 of cmake/get_third_party.cmake to:

`find_package(lion PATHS ${CMAKE_BINARY_DIR}/thirdparty/lib/cmake/lion HINTS ${CMAKE_BINARY_DIR}/thirdparty/lib64/cmake/lion NO_DEFAULT_PATH REQUIRED)`

**Comment by pkanninen** [comment]
Now the cmake ends without errors, log file in attachment. 
[log.log](https://github.com/juanmanzanero/fastest-lap/files/8638724/log.log)


Next issue rise by running make

> [Pekka@Kone7 build]$ make
[  3%] Building CXX object src/main/c/CMakeFiles/fastestlapc.dir/fastestlapc.cpp.o
In file included from /home/Pekka/fastestLap/fastest-lap/src/main/c/fastestlapc.cpp:9:
/home/Pekka/fastestLap/fastest-lap/./src/core/applications/optimal_laptime.h: In instantiation of ‘static Optimal_laptime<Dynamic_model_t>::Control_variable<> Optimal_laptime<Dynamic_model_t>::create_dont_optimize() [with Dynamic_model_t = lot2016kart<CppAD::AD<double> >::Dynamic_model<Road_curvilinear<CppAD::AD<double>, Track_by_polynomial, 10, 2> >]’:
/home/Pekka/fastestLap/fastest-lap/src/main/c/fastestlapc.cpp:1019:117:   required from ‘typename Optimal_laptime<typename vehicle_t::vehicle_ad_curvilinear>::Control_variables<> construct_control_variables(const Optimal_laptime_configuration<vehicle_t>&, std::size_t, const std::array<double, typename vehicle_t::vehicle_ad_curvilinear::NCONTROL>&) [with vehicle_t = lot2016kart_all; typename Optimal_laptime<typename vehicle_t::vehicle_ad_curvilinear>::Control_variables<> = Optimal_laptime<lot2016kart<CppAD::AD<double> >::Dynamic_model<Road_curvilinear<CppAD::AD<double>, Track_by_polynomial, 10, 2> > >::Control_variables<double>; std::size_t = long unsigned int; typename vehicle_t::vehicle_ad_curvilinear = lot2016kart<CppAD::AD<double> >::Dynamic_model<Road_curvilinear<CppAD::AD<double>, Track_by_polynomial, 10, 2> >]’
/home/Pekka/fastestLap/fastest-lap/src/main/c/fastestlapc.cpp:1090:61:   required from ‘void compute_optimal_laptime(vehicle_t&, Track_by_polynomial&, c_Vehicle*, int, const double*, const char*) [with vehicle_t = lot2016kart_all]’
/home/Pekka/fastestLap/fastest-lap/src/main/c/fastestlapc.cpp:1337:64:   required from here
/home/Pekka/fastestLap/fastest-lap/./src/core/applications/optimal_laptime.h:222:118: error: braces around scalar initializer for type ‘Optimal_laptime<lot2016kart<CppAD::AD<double> >::Dynamic_model<Road_curvilinear<CppAD::AD<double>, Track_by_polynomial, 10, 2> > >::Optimal_control_type’
     { return Control_variable<>{ .optimal_control_type{DONT_OPTIMIZE}, .s_hypermesh{}, .u{}, .dudt{}, .dissipation{} }; }
                                                                                                                      ^
compilation terminated due to -Wfatal-errors.
make[2]: *** [src/main/c/CMakeFiles/fastestlapc.dir/build.make:76: src/main/c/CMakeFiles/fastestlapc.dir/fastestlapc.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:253: src/main/c/CMakeFiles/fastestlapc.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

**Comment by juanmanzanero** [comment]
It looks your C++ compiler does not support some of the syntax I use. The GitHub actions are using 9.4.0, consider an upgrade to that version

**Comment by pkanninen** [comment]
By using gcc 11.3 it gives still the same error.

**Comment by juanmanzanero** [comment]
Are you using the latest version of the code? (i.e. can you `git pull`?)

**Comment by pkanninen** [comment]
Yes I used latest version. But now I managed compiled it ! 
Problem was cc links still to old gcc.

One issue rise up during install:

> Install the project...
-- Install configuration: ""
CMake Error at cmake_install.cmake:54 (file):
  file INSTALL cannot find
  "/home/Pekka/fastestLap/fastest-lap/build/thirdparty/lib": No such file or
  directory.

I solve this by renaming lib64 by lib  


Thanks for your effort!
